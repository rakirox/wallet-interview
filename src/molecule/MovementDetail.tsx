import dayjs from 'dayjs';
import React, {PropsWithChildren} from 'react';
import {StyleSheet, View} from 'react-native';
import {Image} from '../atom/Image';
import {Text} from '../atom/Text';
import {Movement} from '../types/Movement';

type Props = PropsWithChildren<{
  movement: Movement;
}>;

export function MovementDetail({movement}: Props): JSX.Element {
  return (
    <View style={styles.container}>
      <View style={styles.shadow}>
        <View style={styles.imageContainer}>
          <Image source={{uri: movement.image}} style={styles.image} />
        </View>
      </View>
      <View style={styles.texts}>
        <Text style={styles.detailText}>Detalles del producto</Text>
        <Text style={styles.purchasedText}>{`Comprado el ${dayjs(
          new Date(movement.createdAt),
        )
          .locale('es')
          .format('DD [de] MMMM, YYYY')}`}</Text>
        <Text style={styles.withText}>Con esta compra acomulaste:</Text>
        <Text style={styles.pointsText}>{movement.points} puntos</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  pointsText: {
    fontSize: 24,
    fontWeight: '800',
  },
  withText: {
    fontSize: 14,
    fontWeight: '800',
    color: '#9B9898',
    marginBottom: 32,
  },
  purchasedText: {
    fontWeight: '800',
    fontSize: 16,
    marginBottom: 20,
  },
  detailText: {
    fontWeight: '800',
    fontSize: 14,
    color: '#9B9898',
    marginBottom: 19,
  },
  texts: {
    paddingHorizontal: 20,
    paddingVertical: 5,
  },
  container: {
    width: '100%',
  },
  shadow: {
    shadowColor: 'rgba(0, 0, 0, 0.5)',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowRadius: 3.84,
    shadowOpacity: 1,
    elevation: 6,
  },
  imageContainer: {
    borderRadius: 20,
    margin: 20,
    overflow: 'hidden',
    height: 350,
  },
  image: {
    flex: 1,
  },
});
