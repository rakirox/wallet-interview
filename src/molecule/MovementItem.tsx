import React, {PropsWithChildren} from 'react';
import {View, StyleSheet, Image, TouchableOpacity} from 'react-native';
import dayjs from 'dayjs';
import {Text} from '../atom/Text';
import {Movement} from '../types/Movement';

type Props = PropsWithChildren<{
  movement: Movement;
  onPress: () => void;
}>;

export function MovmentItem({movement, onPress}: Props): JSX.Element {
  const label = movement.isRedemption ? (
    <Text style={styles.negative}>-</Text>
  ) : (
    <Text style={styles.positive}>+</Text>
  );
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <View style={styles.imageContainer}>
        <Image source={{uri: movement.image}} style={styles.image} />
      </View>
      <View style={styles.textContainer}>
        <Text style={styles.textProduct}>{movement.product}</Text>
        <Text style={styles.textPCreatedAt}>
          {dayjs(new Date(movement.createdAt))
            .locale('es')
            .format('DD [de] MMMM, YYYY')}
        </Text>
      </View>

      <Text style={styles.textPoints}>
        {label}
        {movement.points}
      </Text>

      <View style={styles.arrowContainer}>
        <Image
          source={require('./../assets/images/subtract.png')}
          style={styles.arrow}
        />
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  arrowContainer: {
    // flex: 1,
  },
  arrow: {
    width: 10,
    height: 10,
  },
  negative: {
    color: '#FF0000',
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 16,
    lineHeight: 22,
  },
  positive: {
    color: '#00B833',
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 16,
    lineHeight: 22,
  },
  container: {
    paddingBottom: 20,
    paddingHorizontal: 20,
    flexDirection: 'row',
    alignItems: 'center',
  },
  textContainer: {
    flex: 1,
  },
  textProduct: {
    fontWeight: '800',
    fontSize: 14,
    marginBottom: 7,
  },
  textPoints: {
    fontWeight: '800',
    fontSize: 16,
    lineHeight: 22,
    marginRight: 23,
  },
  textPCreatedAt: {
    fontWeight: '400',
    fontSize: 12,
  },
  image: {flex: 1},
  imageContainer: {
    borderRadius: 10,
    overflow: 'hidden',
    width: 55,
    height: 55,
    marginRight: 11,
  },
});
