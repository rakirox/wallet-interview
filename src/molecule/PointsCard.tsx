import React, {PropsWithChildren} from 'react';
import {ActivityIndicator, StyleSheet, View} from 'react-native';
import {Text} from '../atom/Text';

type Props = PropsWithChildren<{
  title: string;
  points: number;
  loading: boolean;
}>;

export function PointsCard({title, points, loading}: Props): JSX.Element {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      {loading ? (
        <ActivityIndicator style={styles.activityIndicator} color="white" />
      ) : (
        <Text style={styles.points}>
          {`${points.toLocaleString('mxn', {
            minimumFractionDigits: 2,
          })} pts`}
        </Text>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  activityIndicator: {
    alignSelf: 'center',
    marginVertical: 30,
  },
  container: {
    backgroundColor: '#334FFA',
    padding: 20,
    shadowColor: 'rgba(0, 0, 0, 0.5)',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowRadius: 3.84,
    shadowOpacity: 1,
    elevation: 6,
    borderRadius: 20,
    minWidth: 280,
  },
  title: {
    fontWeight: '800',
    fontSize: 16,
    lineHeight: 22,
    color: '#ffffff',
    marginBottom: 7,
  },
  points: {
    fontWeight: '800',
    fontSize: 32,
    lineHeight: 44,
    color: '#ffffff',
    marginBottom: 50,
    alignSelf: 'center',
  },
});
