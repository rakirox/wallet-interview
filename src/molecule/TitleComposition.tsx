import React, {PropsWithChildren} from 'react';
import {StyleSheet} from 'react-native';
import {Text} from '../atom/Text';

type Props = PropsWithChildren<{
  title: string;
  subtitle: string;
}>;

export function TitleComposition({title, subtitle}: Props): JSX.Element {
  return (
    <>
      <Text style={styles.titleWelcome}>{title}</Text>
      <Text style={styles.subtitleWelcome}>{subtitle}</Text>
    </>
  );
}

const styles = StyleSheet.create({
  titleWelcome: {
    fontWeight: '800',
    fontSize: 20,
    lineHeight: 27.32,
    color: '#020202',
  },
  subtitleWelcome: {
    fontWeight: '400',
    fontSize: 16,
    lineHeight: 27.32,
    color: '#020202',
  },
});
