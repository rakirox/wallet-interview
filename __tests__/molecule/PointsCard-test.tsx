/**
 * @format
 */

import 'react-native';
import React from 'react';
import {PointsCard} from '../../src/molecule/PointsCard';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('PointsCard', () => {
  it('render correctly', () => {
    expect(
      renderer.create(
        <PointsCard title="Hello Femsa" points={10000} loading={false} />,
      ),
    ).toMatchSnapshot();
  });

  it('render correctly loading true', () => {
    expect(
      renderer.create(
        <PointsCard title="Hello Femsa" points={10000} loading={true} />,
      ),
    ).toMatchSnapshot();
  });
});
