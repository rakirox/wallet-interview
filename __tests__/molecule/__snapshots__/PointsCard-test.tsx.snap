// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointsCard - render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#334FFA",
      "borderRadius": 20,
      "elevation": 6,
      "minWidth": 280,
      "padding": 20,
      "shadowColor": "rgba(0, 0, 0, 0.5)",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3.84,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "avenir",
        },
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "800",
          "lineHeight": 22,
          "marginBottom": 7,
        },
      ]
    }
  >
    Hello Femsa
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "avenir",
        },
        {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontSize": 32,
          "fontWeight": "800",
          "lineHeight": 44,
          "marginBottom": 50,
        },
      ]
    }
  >
    10,000.00 pts
  </Text>
</View>
`;

exports[`PointsCard - render correctly loading true 1`] = `
<View
  style={
    {
      "backgroundColor": "#334FFA",
      "borderRadius": 20,
      "elevation": 6,
      "minWidth": 280,
      "padding": 20,
      "shadowColor": "rgba(0, 0, 0, 0.5)",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3.84,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "avenir",
        },
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "800",
          "lineHeight": 22,
          "marginBottom": 7,
        },
      ]
    }
  >
    Hello Femsa
  </Text>
  <ActivityIndicator
    color="white"
    style={
      {
        "alignSelf": "center",
        "marginVertical": 30,
      }
    }
  />
</View>
`;
